### PR TITLE
Magazine adaptors [RDY]

### PR DESCRIPTION
**Pull request by mugling**
Requested on a few occasions and implemented because after #16339 I want to start working on generic turrets (#8082).

Turrets should be able to operate autonomously but require a player to reload their ammunition. Converting guns to feed from belts is the logical approach and consistent with reality.

Guns are divided into classes based upon their action as only `rifle_semi` and `rifle_auto` will become eligible for mounting as turrets.

A separate PR will also define a bulky `turret mount` that occupies the `grip` slot as well as an camera-gun sight (name anyone?) occupying `sight` slot to allow for remote/automatic operation.

### CODE REVIEW

**Comment by Rod-Serling** [comment]
> as well as an camera-gun sight (name anyone?) 

https://en.wikipedia.org/wiki/Remote_weapon_station

**Comment by mugling** [comment]
The leadworks magazines are compact and light-weight. ec84d16 adds a rare bootleg adaptor for converting the other guns to use them.

**Comment by mugling** [comment]
Done for now

File: data/json/items/classes/gun.json [hunk 1]
```diff
@@ -4,22 +4,46 @@
...
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "barrel", 1 ], [ "bore", 1 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ], [ "stock", 1 ] ]
   },
   {
     "abstract": "rifle_base",
     "type": "GUN",
     "name": "base rifle",
     "skill": "rifle",
-    "aim_speed": 8
+    "aim_speed": 8,
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "barrel", 1 ], [ "bore", 1 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "magazine", 1 ], [ "muzzle", 1 ], [ "rail", 1 ], [ "sights", 1 ], [ "stock", 1 ], [ "underbarrel", 1 ] ]
```
Comment (Coolthulhu) [inline data/json/items/classes/gun.json]: Moving mods to rifle base will require a bigger review. There are multiple rifles with gimmicks like being unable to use some/all slots.

It would be the best if you split that off into a separate PR. And then checked with some script which rifles change slots and how, because finding them all by hand could take a while.

**Comment by mugling** [comment]
The only consumers are in `223.json` and `308.json`. Is this the only issue with the PR as if so I can quickly confirm the manual check I did with jq?

**Comment by mugling** [comment]
No regressions for any of the affected rifles (steyr_aug, m4a1, hk_g36, h&k416a5, ar15, acr, fn_fal, hk_g3, m134, savage_111f, scar_l)

**Comment by Coolthulhu** [comment]
The leadworks rifles for sure are intended not to have any mods.

**Comment by mugling** [comment]
> The leadworks rifles for sure are intended not to have any mods.

Yes, but they inherit from `l_base_223` not from `rifle_*`

**Comment by Coolthulhu** [comment]
Which in turn inherits from rifle_auto.

**Comment by mugling** [comment]
> Which in turn inherits from rifle_auto.

Yes, but `l_base_223` sets the mod locations

**Comment by mugling** [comment]
Testing with #16378 shows that magazine inheritance was broken affecting not only this PR but also a number of existing items (for example the flintlock carbine). A correction is pushed in 27b221d
